File: 1497. Check If Array Pairs Are Divisible by k/solution.java
class Solution {
    public boolean canArrange(int[] arr, int k) {

        HashMap<Integer, Integer> count = new HashMap<>();

        for (int x : arr) {
            int remainder = ((x % k) + k) % k;
            count.put(remainder, count.getOrDefault(remainder, 0) + 1);
        }


        for (int key : count.keySet()) {

            int freq = count.get(key);

            if (key == 0) {
                if ((freq & 1) != 0) {
                    return false;
                }
            }else {
                int complementFreq = count.getOrDefault(k - key, 0);
                if (freq != complementFreq) {
                    return false;
                }
            }
        }

        return true;

    }
}

File: 1497. Check If Array Pairs Are Divisible by k/solution.py
class Solution:
    def canArrange(self, arr: List[int], k: int) -> bool:
        count = {}

        for x in arr:
            remainder = ((x % k) + k) % k
            count[remainder] = count.get(remainder, 0) + 1

        for key, freq in count.items():
            if key == 0:
                if freq % 2 != 0:
                    return False
            else:
                complement_freq = count.get(k - key, 0)
                if freq != complement_freq:
                    return False

        return True
        